Client/Frontend/Fakespot/Views/FakespotOptInCardView.swift

```diff
@@ -0,0 +1,359 @@
+    @objc
+    private func didTapSecondaryButton() {
+        
+        viewModel?.onTapSecondaryButton()
+    }
+
+    // MARK: View Setup
+    func configure(_ viewModel: FakespotOptInCardViewModel) {
+        self.viewModel = viewModel
+
+        headerLabel.text = viewModel.headerTitleLabel
+        headerLabel.accessibilityIdentifier = viewModel.headerLabelA11yId
+
+        bodyFirstParagraphLabel.attributedText = getFirstParagraphText()
+        bodyFirstParagraphLabel.accessibilityIdentifier = viewModel.bodyFirstParagraphA11yId
+
+        disclaimerTextLabel.attributedText = getDisclaimerText()
+        disclaimerTextLabel.accessibilityIdentifier = viewModel.disclaimerTextLabelA11yId
+
+        learnMoreButton.setTitle(viewModel.learnMoreButtonTitle, for: .normal)
+        learnMoreButton.accessibilityIdentifier = viewModel.learnMoreButtonTitleA11yId
+
+        termsOfUseButton.setTitle(viewModel.termsOfUseButtonTitle, for: .normal)
+        termsOfUseButton.accessibilityIdentifier = viewModel.termsOfUseButtonTitleA11yId
+
+        privacyPolicyButton.setTitle(viewModel.privacyPolicyButtonTitle, for: .normal)
+        privacyPolicyButton.accessibilityIdentifier = viewModel.privacyPolicyButtonTitleA11yId
+
+        mainButton.setTitle(viewModel.mainButtonTitle, for: .normal)
+        mainButton.accessibilityIdentifier = viewModel.mainButtonTitleA11yId
+
+        secondaryButton.setTitle(viewModel.secondaryButtonTitle, for: .normal)
+        secondaryButton.accessibilityIdentifier = viewModel.secondaryButtonTitleA11yId
+
+        let cardModel = ShadowCardViewModel(view: mainView, a11yId: viewModel.cardA11yId)
+        cardContainer.configure(cardModel)
+    }
+
+    func getFirstParagraphText() -> NSAttributedString {
+        let websites = self.getFirstParagraphWebsites()
+        let font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .body,
+                                                          size: UX.bodyFirstParagraphLabelFontSize)
+        let plainText = String.localizedStringWithFormat(viewModel?.bodyFirstParagraphLabel ?? "", websites[0], websites[1], websites[2])
+        return plainText.attributedText(boldStrings: websites, font: font)
+    }
+
+    func getFirstParagraphWebsites() -> [String] {
+        let lowercasedName = self.viewModel?.productSitename?.lowercased() ?? "amazon"
+        var currentPartnerWebsites = ["amazon", "walmart", "bestbuy"]
+
+        // just in case this card will be shown from an unpartnered website in the future
+        guard currentPartnerWebsites.contains(lowercasedName) else {
+            currentPartnerWebsites[2] = "Best Buy"
+            return currentPartnerWebsites.map { $0.capitalized }
+        }
+
+        var websitesOrder = currentPartnerWebsites.filter { $0 != lowercasedName }
+        if lowercasedName == "bestbuy" {
+            websitesOrder.insert("Best Buy", at: 0)
+        } else {
+            websitesOrder.insert(lowercasedName.capitalized, at: 0)
+        }
```

Comment:
This seems to be business logic, it feels strange to have this in the view + could be tested